typescript/cli/src/context/altvm.ts

```diff
@@ -0,0 +1,196 @@
+import { password } from '@inquirer/prompts';
+
+import {
+  CosmosNativeProvider,
+  CosmosNativeSigner,
+} from '@hyperlane-xyz/cosmos-sdk';
+import {
+  ChainMap,
+  ChainMetadataManager,
+  ProtocolMap,
+  isJsonRpcSubmitterConfig,
+} from '@hyperlane-xyz/sdk';
+import { AltVM, MINIMUM_GAS, ProtocolType } from '@hyperlane-xyz/utils';
+
+import { ExtendedChainSubmissionStrategy } from '../submitters/types.js';
+
+import { SignerKeyProtocolMap } from './types.js';
+
+// ### ALL Alt VM PROTOCOLS ARE REGISTERED HERE ###
+const ALT_VM_SUPPORTED_PROTOCOLS: ALT_VM_PROTOCOL = {
+  [ProtocolType.CosmosNative]: {
+    provider: CosmosNativeProvider,
+    signer: CosmosNativeSigner,
+    gas: {
+      CORE_DEPLOY_GAS: (1e6).toString(),
+      WARP_DEPLOY_GAS: (3e6).toString(),
+      TEST_SEND_GAS: (3e5).toString(),
+      AVS_GAS: (3e6).toString(),
+    },
+  },
+  // [NEW PROTOCOL]: {...}
+};
+
+type ALT_VM_PROTOCOL = ProtocolMap<{
+  provider: AltVM.IProviderConnect;
+  signer: AltVM.ISignerConnect;
+  gas: MINIMUM_GAS;
+}>;
+
+class AltVMFactory implements AltVM.IAltVMFactory {
+  public getSupportedProtocols(): ProtocolType[] {
+    return Object.keys(ALT_VM_SUPPORTED_PROTOCOLS) as ProtocolType[];
+  }
+
+  public supports(protocol: ProtocolType) {
+    return !!ALT_VM_SUPPORTED_PROTOCOLS[protocol];
+  }
+
+  public getMinGas(protocol: ProtocolType) {
+    if (!this.supports(protocol)) {
+      throw new Error(`Protocol type ${protocol} not supported in AltVM`);
+    }
+
+    const { gas } = ALT_VM_SUPPORTED_PROTOCOLS[protocol]!;
```

Comment:
why not:
```typescript
const protocolDefinition = ALT_VM_SUPPORTED_PROTOCOLS[protocol];


if (!protocolDefinition) {
      throw new Error(`Protocol type ${protocol} not supported in AltVM`);
    }

return protocolDefinition.gas
```
so you don't have to silence ts errors as it is able to infer that the type is defined after the if branch?